lib/activejob-status/storage.rb

```diff
@@ -26,6 +26,10 @@ def read(job)
         store.read(key(job)) || {}
       end
 
+      def read_multi(jobs)
+        store.read_multi(*jobs.map { |job| key(job) })
+      end
+
```

Comment:
> [...] However, if there's no cache hit, that cache key isn't included in the result hash [...]

`Storage#read` always return an Hash, even on a cache miss.

```ruby
def read(job)
  store.read(key(job)) || {}
end
```

For consistency, I would recommend `Storage#read_multi` to always return an Hash for every job :

```ruby
def read_multi(jobs)
  keys = jobs.map { |job| key(job) }
  data = store.read_multi(*keys)
  keys.index_with { |k| data.fetch(k, {}) }
end

Storage.read_multi([job1, job2, job3])
# { "job1" => { status: 'finished' },"job2" => {}, "job3" => { status: 'finished' }]
```

We could then simplify `Batch` implementation and ease further evolutions (see above)

Since the method now has a different I/O API than ActiveSupport's '#read_multi', I would also recommend using a different name to avoid exceptions due to Rails habits.

Why not `read_all(jobs)` ?

Reply:
I implemented your suggestions except the method rename here.

I don't have any specific opinions on the rename - it's used exclusively internally inside of this library. The `#read` method, also has a different return signature than the rails cache adapter (which returns `nil` on a cache miss, instead of an empty hash).